sqlrunner/main.py

```diff
@@ -87,10 +89,18 @@ def write_results(results, f_path):
         return
 
     fieldnames = first_result.keys()
-    with f_path.open(mode="w", newline="", encoding="utf-8") as f:
+
+    if f_path.suffixes == [".csv", ".gz"]:
+        f = gzip.open(f_path, "wt", newline="", compresslevel=6)
```

Comment:
does this need explicit `encoding="utf-8"`, the same as the other branch?

Reply:
The t in "wt" implies encoding in utf-8, but its probably better to be explicit

Reply:
ah